plugins/search-backend-module-techdocs/src/collators/DefaultTechDocsCollatorFactory.ts

```diff
@@ -177,81 +185,83 @@ export class DefaultTechDocsCollatorFactory implements DocumentCollatorFactory {
-          limit(async (): Promise<TechDocsDocument[]> => {
-            const entityInfo =
-              DefaultTechDocsCollatorFactory.handleEntityInfoCasing(
-                this.legacyPathCasing,
-                {
-                  kind: entity.kind,
-                  namespace: entity.metadata.namespace || 'default',
-                  name: entity.metadata.name,
-                },
-              );
+      const filteredEntities = this.entityFilterFunction
+        ? this.entityFilterFunction(entities)
+        : this.defaultFilteringFunction(entities);
 
-            try {
-              const { token: techdocsToken } =
-                await this.auth.getPluginRequestToken({
-                  onBehalfOf: await this.auth.getOwnServiceCredentials(),
-                  targetPluginId: 'techdocs',
-                });
+      const docPromises = filteredEntities.map((entity: Entity) =>
+        limit(async (): Promise<TechDocsDocument[]> => {
+          const entityInfo =
+            DefaultTechDocsCollatorFactory.handleEntityInfoCasing(
+              this.legacyPathCasing,
+              {
+                kind: entity.kind,
+                namespace: entity.metadata.namespace || 'default',
+                name: entity.metadata.name,
+              },
+            );
 
-              const searchIndexResponse = await fetch(
-                DefaultTechDocsCollatorFactory.constructDocsIndexUrl(
-                  techDocsBaseUrl,
-                  entityInfo,
-                ),
-                {
-                  headers: {
-                    Authorization: `Bearer ${techdocsToken}`,
-                  },
+          try {
+            const { token: techdocsToken } =
+              await this.auth.getPluginRequestToken({
+                onBehalfOf: await this.auth.getOwnServiceCredentials(),
+                targetPluginId: 'techdocs',
+              });
+
+            const searchIndexResponse = await fetch(
+              DefaultTechDocsCollatorFactory.constructDocsIndexUrl(
+                techDocsBaseUrl,
+                entityInfo,
+              ),
+              {
+                headers: {
+                  Authorization: `Bearer ${techdocsToken}`,
                 },
-              );
+              },
+            );
 
-              // todo(@backstage/techdocs-core): remove Promise.race() when node-fetch is 3.x+
-              // workaround for fetch().json() hanging in node-fetch@2.x.x, fixed in 3.x.x
-              // https://github.com/node-fetch/node-fetch/issues/665
-              const searchIndex = await Promise.race([
-                searchIndexResponse.json(),
-                new Promise((_resolve, reject) => {
-                  setTimeout(() => {
-                    reject('Could not parse JSON in 5 seconds.');
-                  }, 5000);
-                }),
-              ]);
+            // todo(@backstage/techdocs-core): remove Promise.race() when node-fetch is 3.x+
```

Comment:
Not related to this change but is this still needed as the `node-fetch` was removed? @freben ?